File: mycollab-web/src/main/java/com/esofthead/mycollab/module/project/ui/events/ProjectEvent.java
package com.esofthead.mycollab.module.project.ui.events;

import com.esofthead.mycollab.vaadin.events.ApplicationEvent;

@SuppressWarnings("serial")
public class ProjectEvent {

	public static class Save extends ApplicationEvent {
		public Save(Object source, Object data) {
			super(source, data);
		}
	}
	
	public static class GetMyProjects extends ApplicationEvent {
		public GetMyProjects(Object source, Object data) {
			super(source, data);
		}
	}
	
	public static class GotoMyProject extends ApplicationEvent {
		public GotoMyProject(Object source, Object data) {
			super(source, data);
		}
	}

	public static class GotoMyProjectList extends ApplicationEvent {
		public GotoMyProjectList(Object source, Object data) {
			super(source, data);
		}
	}
}

File: mycollab-mobile/src/main/scala/com/esofthead/mycollab/mobile/module/project/events/ProjectEvent.scala
package com.esofthead.mycollab.mobile.module.project.events

import com.esofthead.mycollab.eventmanager.ApplicationEvent

/**
  * @author MyCollab Ltd
  * @since 5.2.5
  */
object ProjectEvent {

  class GotoAdd(source: AnyRef, data: AnyRef) extends ApplicationEvent(source, data) {}

  class GotoProjectList(source: AnyRef, data: AnyRef) extends ApplicationEvent(source, data) {}

  class GotoMyProject(source: AnyRef, data: AnyRef) extends ApplicationEvent(source, data) {}

  class MyProjectActivities(source: AnyRef, data: Integer) extends ApplicationEvent(source, data) {}

  class AllActivities(source: AnyRef) extends ApplicationEvent(source, null) {}

}
